gensim/test/test_ldamodel_distributed.py

```diff
@@ -0,0 +1,527 @@
+
+        kwargs['alpha'] = [[0.3], [0.3]]
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['alpha'] = [0.3]
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['alpha'] = "gensim is cool"
+        self.assertRaises(ValueError, self.class_, **kwargs)
+
+    def testEtaAuto(self):
+        model1 = self.class_(corpus, id2word=dictionary, eta='symmetric', passes=10)
+        modelauto = self.class_(corpus, id2word=dictionary, eta='auto', passes=10)
+
+        # did we learn something?
+        self.assertFalse(np.allclose(model1.eta, modelauto.eta))
+
+    def testEta(self):
+        kwargs = dict(
+            id2word=dictionary,
+            num_topics=2,
+            eta=None
+        )
+        num_terms = len(dictionary)
+        expected_shape = (num_terms,)
+
+        # should not raise anything
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.5] * num_terms))
+
+        kwargs['eta'] = 'symmetric'
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.5] * num_terms))
+
+        kwargs['eta'] = 0.3
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.3] * num_terms))
+
+        kwargs['eta'] = 3
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([3] * num_terms))
+
+        kwargs['eta'] = [0.3] * num_terms
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.3] * num_terms))
+
+        kwargs['eta'] = np.array([0.3] * num_terms)
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.3] * num_terms))
+
+        # should be ok with num_topics x num_terms
+        testeta = np.array([[0.5] * len(dictionary)] * 2)
+        kwargs['eta'] = testeta
+        self.class_(**kwargs)
+
+        # all should raise an exception for being wrong shape
+        kwargs['eta'] = testeta.reshape(tuple(reversed(testeta.shape)))
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['eta'] = [0.3]
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['eta'] = [0.3] * (num_terms + 1)
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['eta'] = "gensim is cool"
+        self.assertRaises(ValueError, self.class_, **kwargs)
+
+        kwargs['eta'] = "asymmetric"
+        self.assertRaises(ValueError, self.class_, **kwargs)
+
+    def testTopTopics(self):
+        top_topics = self.model.top_topics(self.corpus)
```

Comment:
Why are you checking the type of top_top topics and its keys, instead of checking the type and actual value? Why not do something like:

```
expected_top_topics = [
    ('foo', 'bar'),
    ...,
]
self.assertEqual(top_topics, expected_top_topics)
```

That would be a more robust and easy-to-read test, in my opinion.

The same thing applies to other tests as well.